packages/react/src/components/Popover/Popover.tsx

```diff
@@ -1,256 +1,151 @@
-
-interface PopoverTriggerProps {
-  children: React.ReactNode;
-}
-
-const PopoverTrigger = forwardRef<
-  HTMLElement,
-  React.HTMLProps<HTMLElement> & PopoverTriggerProps
->(function PopoverTrigger({ children, ...props }, propRef) {
-  const context = usePopoverContext();
-
-  const child = isValidElement(children)
-    ? (children as React.ReactElement & React.RefAttributes<HTMLElement>)
-    : null;
-  const ref = useMergeRefs([context.refs.setReference, propRef]);
-
-  if (child) {
-    const childProps = {
-      ref,
-      ...props,
-      ...(child.props as Record<string, unknown>),
-      ...context.getReferenceProps(),
-      'data-state': context.open ? 'open' : 'closed',
-      'aria-expanded': context.open,
-    };
-
-    return cloneElement(child, childProps);
-  }
-
-  return null;
-});
-
-const PopoverContent = forwardRef<
-  HTMLDivElement,
-  React.HTMLProps<HTMLDivElement>
->(function PopoverContent(props, propRef) {
-  const context = usePopoverContext();
-  const ref = useMergeRefs([context.refs.setFloating, propRef]);
-
-  return context.open ? (
-    <div
-      ref={ref}
-      style={{
-        position: context.strategy,
-        top: context.y ?? 0,
-        left: context.x ?? 0,
-      }}
-      data-placement={context.placement}
-      className={cn(
-        classes.popover,
-        classes[context.variant],
-        context.className,
-      )}
-      {...context.getFloatingProps(props)}
-      tabIndex={-1}
-      role={context.role || 'dialog'}
-    >
-      {props.children}
-    </div>
-  ) : null;
-});
-
-const PopoverArrow = forwardRef<
-  HTMLDivElement,
-  React.HTMLProps<HTMLDivElement>
->(function PopoverContent(props, propRef) {
-  const context = usePopoverContext();
-  const ref = useMergeRefs([context.arrowRef, propRef]);
-
-  const arrowX = context.middlewareData.arrow?.x;
-  const arrowY = context.middlewareData.arrow?.y;
-
-  // Get the placement of the popover arrow independent of alignment, which is opposite of popover content placement.
-  // Used to align the arrow to the edge of the content.
-  const staticSide: string | undefined = {
-    top: 'bottom',
-    right: 'left',
-    bottom: 'top',
-    left: 'right',
-  }[context.placement.split('-')[0]];
-
-  return (
-    <div
-      ref={ref}
-      style={{
-        ...(arrowX != null ? { left: arrowX } : {}),
-        ...(arrowY != null ? { top: arrowY } : {}),
-        ...(staticSide ? { [staticSide]: '-7px' } : {}),
-      }}
-      className={classes.arrow}
-      {...props}
-    />
-  );
-});
+import type { Placement } from '@floating-ui/react';
+import {
+  useFloating,
+  offset,
+  shift,
+  flip,
+  autoUpdate,
+  useClick,
+  useFocus,
+  useDismiss,
+  arrow,
+  useInteractions,
+  useMergeRefs,
+  useRole,
+} from '@floating-ui/react';
+import type { HTMLAttributes } from 'react';
+import React, { forwardRef, useLayoutEffect, useRef } from 'react';
+import cn from 'classnames';
+
+import styles from './Popover.module.css';

```

Comment:
Suggest renaming `styles` to `classes` to be in line with naming in our other components.